lib/render/SliceRenderer.cpp

```diff
@@ -0,0 +1,374 @@
+    _textureHeight = 250;
+
+    SliceParams* p = dynamic_cast<SliceParams*>(GetActiveParams());
+    assert(p);
+    MapperFunction* tf = p->GetMapperFunc(_cacheParams.varName);
+
+    _colorMapSize = tf->getNumEntries();
+    _colorMap = new GLfloat[_colorMapSize * 4];
+    for (int i=0; i<_colorMapSize; i++) {
+        _colorMap[i*4+0] = (float) i / (float) (_colorMapSize-1);
+        _colorMap[i*4+1] = (float) i / (float) (_colorMapSize-1);
+        _colorMap[i*4+2] = (float) i / (float) (_colorMapSize-1);
+        _colorMap[i*4+3] = 1.0;
+    }
+
+    _textureData = new unsigned char[_textureWidth * _textureHeight * 4];
+    _saveCacheParams();
+    _initialized = true;
+}
+
+SliceRenderer::~SliceRenderer() {
+    glDeleteTextures(1, &_texture);
+    if (_colorMap) delete [] _colorMap;
+    if (_textureData) delete [] _textureData;
+}
+
+void SliceRenderer::_initTexture() {
+    glDeleteTextures(1, &_texture);
+    glGenTextures(1, &_texture);
+
+    glBindTexture(GL_TEXTURE_2D, _texture);
+    glPolygonMode(GL_FRONT_AND_BACK,GL_FILL);
+    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+    glEnable(GL_BLEND);
+
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_REPEAT);	
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_REPEAT);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+    //glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+    //glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+
+    glTexImage2D(
+        GL_TEXTURE_2D, 0, GL_RGBA, _textureWidth, _textureHeight, 
+        0, GL_RGBA, GL_UNSIGNED_BYTE, (GLvoid*)_textureData
+    );
+}
+
+void SliceRenderer::_saveCacheParams() {
+    SliceParams* p = dynamic_cast<SliceParams*>(GetActiveParams());
+    assert(p);
+
+    _cacheParams.varName            = p->GetVariableName();
+    _cacheParams.heightVarName      = p->GetHeightVariableName();
+    _cacheParams.ts                 = p->GetCurrentTimestep();
+    _cacheParams.refinementLevel    = p->GetRefinementLevel();
+    _cacheParams.compressionLevel   = p->GetCompressionLevel();
+    _cacheParams.textureSampleRate  = p->GetSampleRate();
+    _cacheParams.orientation        = p->GetBox()->GetOrientation();
+
+    _textureWidth  = _cacheParams.textureSampleRate;
+    _textureHeight = _cacheParams.textureSampleRate;
+    if (_textureWidth > MAXTEXTURESIZE)
+        _textureWidth = MAXTEXTURESIZE;
+    if (_textureHeight > MAXTEXTURESIZE)
+        _textureHeight = MAXTEXTURESIZE;
+ 
+    p->GetBox()->GetExtents(_cacheParams.boxMin, _cacheParams.boxMax);
+
+    MapperFunction* tf = p->GetMapperFunc(_cacheParams.varName);
+    tf->makeLut(_cacheParams.tf_lut);
+    _cacheParams.tf_minMax = tf->getMinMaxMapValue();
+    
+    if (_textureData) delete [] _textureData;
+    _textureData = new unsigned char[_textureWidth * _textureHeight * 4];
+
+    int rc = _saveTextureData();
```

Comment:
If saveTextureData() generates an error condition that calls SetErrMsg() then saveCacheParams should pass the error upwards (it can't return void).